firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -119,11 +122,19 @@ public Thread newThread(Runnable r) {
       RandomFidGenerator fidGenerator) {
     this.firebaseApp = firebaseApp;
     this.serviceClient = serviceClient;
-    this.executor = executor;
     this.persistedInstallation = persistedInstallation;
     this.utils = utils;
     this.iidStore = iidStore;
     this.fidGenerator = fidGenerator;
+    this.backgroundExecutor = backgroundExecutor;
+    this.networkExecutor =
+        new ThreadPoolExecutor(
+            /* corePoolSize= */ CORE_POOL_SIZE,
+            /* maximumPoolSize= */ MAXIMUM_POOL_SIZE,
+            /* keepAliveTime= */ 0L,
+            TimeUnit.MILLISECONDS,
+            /* workQueue= */ new LinkedBlockingQueue<>(),
+            /* threadFactory= */ THREAD_FACTORY);
```

Comment:
No comments needed since you are using constants that explain the purpose of the parameters.